### PR TITLE
Feature/add autofix function

### PR DESCRIPTION
**Pull request by imvetri**

### CODE REVIEW

**Comment by AndyOGo** [comment]
@imvetri 
Cool, thanks for your effort, very appreciated.
I made some slight adjustments, so that linting passes.

I only see one missing part, updating the readme.

**Comment by imvetri** [comment]
@AndyOGo  Updated docs. Need some scenario samples for writing the test, could you please share any?

**Comment by AndyOGo** [comment]
@imvetri 
Very cool.
I'm sure you found already where the [tests are](https://github.com/AndyOGo/stylelint-declaration-strict-value/tree/master/test).

I think we should just concentrate on the generic functionality and make sure that autofixing itself is working

File: README.md [hunk 1]
```diff
@@ -290,6 +293,36 @@ a { color: @color-white; }
...
+
+Autofixing the rule violation can be acheived using a custom function, like:
+
+```js
+// .stylelintrc
+"rules": {
+  // ...
+  "scale-unlimited/declaration-strict-value": ["/color/", {
+    disableFix: true,
+    autofixFunc: function(root, node, context){ /* css AST change code goes here */ }
```
Comment (AndyOGo) [inline README.md]: @imvetri 
I just realisied we still should pass which type broke a rule, i.e. `validVar`, `validFunc`, `validKeyword`

**Comment by AndyOGo** [comment]
@imvetri 
The docs should also note that the js config is required for autofixing.

And I'm thinking that it would be better to have the scheme section rather at the beginning then the end 🤔

**Comment by imvetri** [comment]
@AndyOGo great callout! regarding the scheme. Lets take it to top?

File: README.md [hunk 2]
```diff
@@ -607,11 +607,12 @@ a {
 }
 ```
 
-### Scheme
+### Scheme - No autofix
```
Comment (AndyOGo) [inline README.md]: @imvetri 
Sorry, the **Scheme** is intended a single quick overview listing all possible values at once.
May I ask you to undo the this last change?

Comment (imvetri) [reply README.md]: Done!

**Comment by imvetri** [comment]
@AndyOGo I tried to look in stylelint repo for some sample tests against fixes unfortunately I couldn't spot the right one. If you come across any reference please share link to lines of code that tests code fixes.

File: test/auto-fix-function.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  ruleName,
+  skipBasicChecks: true,
+
+  config: [
+    ["/margin/", "/padding/"], {
+    "disableFix": false,
+    "autofixFunc": function(){
+      console.log("lalalal");
+    },
+    fix: true
```
Comment (imvetri) [inline test/auto-fix-function.js]: @AndyOGo I am blocked on writing unit test. When I pass `fix` in the config test I was hoping to get the `context.fix` available at [line104](https://github.com/AndyOGo/stylelint-declaration-strict-value/pull/20/files#diff-1fdf421c05c1140f6d71444ea2b27638R104) but it comes as undefined. 

Do you have any suggestion how to proceed with this?

Comment (AndyOGo) [reply test/auto-fix-function.js]: @imvetri 
That is weird.

I just searched the official stylelint repo for their **autofixing** unit tests and I found this example:
https://github.com/stylelint/stylelint/blob/master/system-tests/fix/fix.test.js

**Comment by AndyOGo** [comment]
@imvetri 
I haven't heard back from you.
So I started a separate branch and implemented it for you #29

**Pull request closed without merging**